lib/Sema/TypeCheckProtocol.cpp

```diff
@@ -7146,20 +7131,17 @@ void TypeChecker::inferDefaultWitnesses(ProtocolDecl *proto) {
       continue;
     }
 
-    if (isa<TypeDecl>(valueDecl))
-      continue;
-
-    if (!valueDecl->isProtocolRequirement())
+    if (isa<TypeDecl>(requirement))
```

Comment:
Change this to an assert since you've already handled associated types above and we don't have other kinds of type requirements

Reply:
Done!